static/app/utils/useKeyPress.tsx

```diff
@@ -1,33 +1,39 @@
-import {useEffect, useState} from 'react';
+import {RefObject, useEffect, useState} from 'react';
 
 /**
  * Hook to detect when a specific key is being pressed
  */
-function useKeyPress(targetKey: React.KeyboardEvent['key']) {
+const useKeyPress = (targetKey: string, ref?: RefObject<HTMLInputElement>) => {
   const [keyPressed, setKeyPressed] = useState(false);
 
-  useEffect(() => {
-    function downHandler({key}: KeyboardEvent) {
-      if (key === targetKey) {
-        setKeyPressed(true);
-      }
+  const downHandler = ({key}: KeyboardEvent) => {
+    if (key === targetKey) {
+      setKeyPressed(true);
     }
+  };
 
-    function upHandler({key}: KeyboardEvent) {
-      if (key === targetKey) {
-        setKeyPressed(false);
-      }
+  const upHandler = ({key}: KeyboardEvent) => {
+    if (key === targetKey) {
+      setKeyPressed(false);
     }
-    window.addEventListener('keydown', downHandler);
-    window.addEventListener('keyup', upHandler);
+  };
+
+  useEffect(() => {
+    const current = ref?.current || window;
+    // @ts-ignore
```

Comment:
What are you ignoring?

Reply:
```
Argument of type '({ key }: KeyboardEvent) => void' is not assignable to parameter of type 'EventListenerOrEventListenerObject'.
  Type '({ key }: KeyboardEvent) => void' is not assignable to type 'EventListener'.
    Types of parameters '__0' and 'evt' are incompatible.
      Type 'Event' is missing the following properties from type 'KeyboardEvent': altKey, charCode, code, ctrlKey, and 17 more.ts(2345)
```

🤷 I suppose ts doesn't like the `||`, but is happy with either window OR ref.current individually

actually - i can rewrite this so that we don't need to ignore 👍

Reply:
Yeah I would strongly try and avoid a `@ts-ignore`